multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -147,12 +147,22 @@ func (c *Cluster) Create(stackBody string) error {
 		return err
 	}
 
+	tags := make([]*cloudformation.Tag, len(c.StackTags))
+	i := 0
+	for k, v := range c.StackTags {
+		key := k
+		value := v
```

Comment:
You need to take the reference of the variables, and loops in golang re-use
variables across iterations. Need to copy them in this case
On Apr 15, 2016 10:43 AM, "Curtis Gagliardi" <notifications@github.com>
wrote:

In multi-node/aws/pkg/cluster/cluster.go
<https://github.com/coreos/coreos-kubernetes/pull/406#discussion_r59912137>:

> @@ -147,12 +147,22 @@ func (c *Cluster) Create(stackBody string) error {
>  		return err
>  	}
>
> +	tags := make([]*cloudformation.Tag, len(c.StackTags))
> +	i := 0
> +	for k, v := range c.StackTags {
> +		key := k
> +		value := v

Why not for key, value := range c.StackTags {? Or you could just use k and
v below. I'm generally not a fan of introducing variables unless they're
necessary.

Similarly I'd prefer to drop i and instead write tags = append(tags,
&cloudFormation..Tag{...}.

Otherwise this looks great to me, thanks :)

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/coreos/coreos-kubernetes/pull/406/files/6442f35d68173d513c36e8be8c6f45e5cd930287#r59912137>